src/bridge/cockpitpipechannel.c

```diff
@@ -105,6 +108,38 @@ process_pipe_buffer (CockpitPipeChannel *self,
 }
 
 static gboolean
+cockpit_pipe_channel_read_window_size_options (JsonObject *options,
+                                               gushort default_rows,
+                                               gushort default_cols,
+                                               gushort *rowsp,
+                                               gushort *colsp)
+{
+  gint64 rows, cols;
+  JsonObject *window;
+
+  if (!cockpit_json_get_object (options, "window", NULL, &window))
+    return FALSE;
+
+  if (window == NULL)
+    {
+      *rowsp = default_rows;
+      *colsp = default_cols;
+      return TRUE;
+    }
+
+  if (cockpit_json_get_int (window, "rows", default_rows, &rows) &&
+      cockpit_json_get_int (window, "cols", default_cols, &cols) &&
+      rows >= 0 && rows <= G_MAXUINT16 && cols >= 0 && cols <= G_MAXUINT16)
```

Comment:
I know it's a long shot, but the front end doesn't seem to limit itself to G_MAXUINT16? How about just clamping it to these maximums, rather than disconnecting the front end for passing invalid data?